app/src/stores/post.ts

```diff
@@ -37,27 +59,53 @@ export const usePostStore = defineStore("post", () => {
      * Get all posts by tag with optional sorting.
      */
     const postsByTag = computed(() => {
-        return (tagId: Uuid, options: postSortOptions | undefined) => {
-            const res = posts.value?.filter((p) => p.tags.some((t) => t._id == tagId));
-            if (options && options.sortBy) {
-                if (options.sortBy == "publishDate") {
-                    const sorted = res?.sort((a, b) => {
+        return (tagId: Uuid, queryOptions?: postQueryOptions) => {
+            // query
+            let res = posts.value?.filter((p) => {
+                return p.tags.some((t) => t._id == tagId);
+            });
+
+            // optional sorting
+            if (
+                res &&
```

Comment:
Maybe just have an early return so you don't have to check if `res` exists for every `if`. Also this method now returns either an array or undefined, not sure if that's intended or if an empty array would make sense as a default.